### DISCUSSION TITLE
Two different way of getting Directus from Docker but only one is working

### DISCUSSION THREAD

**Discussion by suhaimim** [directus/directus#9941] [Archive]
This way of getting docker repo for directus is not working (unable to open CLI nor Browser)

![image](https://user-images.githubusercontent.com/35832684/142722463-4382a9bd-fd1a-4cc0-832c-08ac56e4de82.png)

This way of getting docker repo for directus is running fine

![image](https://user-images.githubusercontent.com/35832684/142722537-636577a7-fe99-4416-9eb5-ea2de6dfef31.png)

**Comment by rijkvanzanten** [comment]
I'm not sure what you mean.. do you mean that "docker pull" doesn't run anything? If so, that's correct. Pull is to download the image, run is to run it 🤔

**Reply by suhaimim** [reply]
Yes, your understand is half correct.
Docker pull did the pull the Directus image.
But once its complete, suppose there will be these five icons
![image](https://user-images.githubusercontent.com/35832684/142743319-86896c06-7bee-48be-b1d5-3e63eb3a697a.png)
But, in this case, the first icon was not there.
When I click run icon, it just execute for a few seconds and then stop by itself.

**Reply by azrikahar** [reply]
I think this is more of a Docker Desktop (assuming from your screenshot) question, rather than a Directus related thing. 

> When I click run icon, it just execute for a few seconds and then stop by itself.

Disclaimer: Not super familiar with Docker Desktop, just more comfortable with Docker via cli

I believe what you're experiencing here is when you click run, it's as if it's simply running `docker run directus/directus:latest`. This fails because Directus requires `KEY` and `SECRET` env variables set, hence it "stop by itself". Using the "different" way which adds the 2 env variables in the command works, but once again this is not actually two different way, as they are both running the same docker run command at the end of the day, just one without extra env vars as you're using the GUI. This scenario should hold true even if we were to use any other images that has required env variables, not just Directus. 👍

**Reply by suhaimim** [reply]
I c. Ok, understand.
Thank you for the clarification.
Usually docker image that able to pull directly already has their own env config.
During local run, just provide the parameters to make it run. So I tought that Directus docker image that able to pull directly was similar to the understanding.
